cpp/tests/core/Tensor.cpp

```diff
@@ -2801,5 +2801,46 @@ TEST_P(TensorPermuteDevices, NumpyIO) {
     utility::filesystem::RemoveFile(file_name);
 }
 
+TEST_P(TensorPermuteDevices, RValueScalar) {
+    const core::Device &device = GetParam();
+    core::Tensor t, t1;
+
+    // Assign scalars basic tests.
+    t = core::Tensor::Init<uint8_t>({{1, 2, 3, 4}, {5, 6, 7, 8}}, device);
+    t1 = core::Tensor::Init<uint8_t>({{1, 10, 3, 4}, {5, 6, 7, 8}}, device);
+    t[0][1] = 10;
+    EXPECT_EQ(t.GetShape(), t1.GetShape());
+    EXPECT_EQ(t.ToFlatVector<uint8_t>(), t1.ToFlatVector<uint8_t>());
+
+    t = core::Tensor::Init<uint8_t>({{1}, {2}}, device);
+    t1 = core::Tensor::Init<uint8_t>({{0}, {4}}, device);
+    t[0][0] = 0;
+    t[1][0] = 4;
+    EXPECT_EQ(t.GetShape(), t1.GetShape());
+    EXPECT_EQ(t.GetDtype(), t1.GetDtype());
+    EXPECT_EQ(t.ToFlatVector<uint8_t>(), t1.ToFlatVector<uint8_t>());
+
+    // Check with boolean.
+    t = core::Tensor::Init<bool>({false}, device);
+    t1 = core::Tensor::Init<bool>({true}, device);
+    t[0] = true;
+    EXPECT_EQ(t.GetShape(), t1.GetShape());
+    EXPECT_EQ(t.ToFlatVector<bool>(), t1.ToFlatVector<bool>());
+
+    // Check with Slice.
+    t = core::Tensor::Init<uint8_t>({1}, device);
+    t1 = core::Tensor::Init<uint8_t>({255}, device);
+    t.Slice(0, 0, 1) = 255;
+    EXPECT_EQ(t.GetShape(), t1.GetShape());
+    EXPECT_EQ(t.ToFlatVector<uint8_t>(), t1.ToFlatVector<uint8_t>());
+
+    // Datatype implicit conversion with Slice.
+    t = core::Tensor::Init<bool>({false}, device);
+    t1 = core::Tensor::Init<bool>({true}, device);
+    t.Slice(0, 0, 1) = 1;
+    EXPECT_EQ(t.GetShape(), t1.GetShape());
+    EXPECT_EQ(t.ToFlatVector<bool>(), t1.ToFlatVector<bool>());
```

Comment:
Please add test cases for tensor with one element (shape `{}`, `{1}`, `{1, 1}`) and without `Slice()`.

Reply:
Done.

Reply:
Done